datafusion/physical-expr/src/expressions/case.rs

```diff
@@ -1269,6 +1312,28 @@ impl CaseExpr {
             self.body.expr_or_expr(batch, when_value)
         }
     }
+
+    fn with_lookup_table(
+        &self,
+        batch: &RecordBatch,
+        scalars_or_null_lookup: &LiteralLookupTable,
+    ) -> Result<ColumnarValue> {
+        let expr = self.body.expr.as_ref().unwrap();
+        let evaluated_expression = expr.evaluate(batch)?;
+
+        let is_scalar = matches!(evaluated_expression, ColumnarValue::Scalar(_));
+        let evaluated_expression = evaluated_expression.to_array(1)?;
+
+        let output = scalars_or_null_lookup.evaluate_input(&evaluated_expression)?;
```

Comment:
Still struggling a bit with this line. It doesn't really read like 'perform lookup in table'. `lookup_table.get(...)` similar to `HashMap::get` would seem to be better describe what we're doing here.

Reply:
changed to `map_input_to_output`